sys/net/gnrc/network_layer/ipv6/nib/nib.c

```diff
@@ -256,22 +256,18 @@ void gnrc_ipv6_nib_handle_timer_event(void *ctx, uint16_t type)
 static size_t _get_l2src(kernel_pid_t iface, uint8_t *l2src,
                          size_t l2src_maxlen)
 {
-    bool try_long = false;
-    int res;
     uint16_t l2src_len;
     /* maximum address length that fits into a minimum length (8) S/TL2A
      * option */
     const uint16_t max_short_len = 6;
 
-    /* try getting source address */
-    if ((gnrc_netapi_get(iface, NETOPT_SRC_LEN, 0, &l2src_len,
-                         sizeof(l2src_len)) >= 0) &&
-        (l2src_len > max_short_len)) {
-        try_long = true;
-    }
-
-    if (try_long && ((res = gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG, 0,
-                                            l2src, l2src_maxlen)) > max_short_len)) {
+    /* get source address length */
+    int res = gnrc_netapi_get(iface, NETOPT_SRC_LEN, 0,
+                              &l2src_len, sizeof(l2src_len));
+    /* get source address, try long address first if possible  */
+    if ((res >= 0) && (l2src_len > max_short_len) &&
+        (gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG, 0,
```

Comment:
Here you changed the behavior, the previous code, set `res` to the result of `gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG,..)`, here it gets the result of `gnrc_netapi_get(iface, NETOPT_SRC_LEN,..)`.

Reply:
I changed the assignment of `res` (that's right) and I also removed `try_long`, *but* did I really changed the behaviour or result of the function? Please also note: `res` isn't used afterwards anyway, the return value of this function was and still is `l2src_len`.

Reply:
I cannot see this change anymore, did you remove it ?
What I see on the original code, there is `l2src_len = (uint16_t)res;` just after this patch and so the return value would have taken the value of `res = gnrc_netapi_get(iface, NETOPT_ADDRESS_LONG,)` which is not the case anymore.

Reply:
badly rebased, and thereby dropped some commit, I reintroduced them - so you should see them again. But also: I still don't think I changed the behaviour of the function - just reorganised and optimised the code a bit.

Reply:
The last line should be `if ((res >= 0) && (l2src_len > max_short_len) && (res = gnrc_net_api_get(iface,NETOPT_ADDRESS_LONG, ...) > (int)max_short_len))` with `res =` to be the same behaviour as the next line is `l2src_len = (uint16_t)res;`

Reply:
you're right, fixed - below too